### PR TITLE
Add school selection page

### LINKED ISSUE #2936: SchoolSelectionPage component
https://app.moqups.com/greg@adhocteam.us/l32hJ36b/view/page/a58f38494

### PR DESCRIPTION
**Pull request by jbalboni**
Closes #2936.

Adds the school selection page. One thing to note is that partial addresses are allowed through. The Address component needs to be updated to handle that, so I've created an issue for it.

### CODE REVIEW

**Comment by jbalboni** [comment]
@akainic Can you take a look?

**Comment by jbalboni** [comment]
screen shot 2016-09-07 at 11 35 03 am

File: src/js/edu-benefits/components/SchoolSelectionFields.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            label="Type of education or training"
+            name="educationType"
+            options={schoolTypes}
+            value={this.props.data.educationType}
+            onValueChange={(update) => {this.props.onStateChange('educationType', update);}}/>
+      </div>
+      <div className="input-section">
+          {showSchoolAddress(this.props.data.educationType.value)
+          ? <div>
+            <p>Do you know what school you will be attending?</p>
```
Comment (akainic) [inline src/js/edu-benefits/components/SchoolSelectionFields.jsx]: Is this a question? Seems strange to have a question that doesn't correspond to a field. Probably a question for @cweber.

Comment (jbalboni) [inline src/js/edu-benefits/components/SchoolSelectionFields.jsx]: Yeah, it's sort of a heading for a block of fields. Maybe it should be styled differently that what I have.

Comment (akainic) [inline src/js/edu-benefits/components/SchoolSelectionFields.jsx]: Not a blocker for getting this merged in (might be fastest to review all these things once we have the complete form), but just wanted to point it out to return to later.

File: src/js/edu-benefits/utils/options-for-select.js [hunk 2]
```diff
@@ -1595,6 +1595,16 @@ const employmentPeriodTiming = [
   { label: 'After military service', value: 'after' }
 ];
 
+const schoolTypes = [
+  { label: 'College or other school', value: 'college' },
+  { label: 'Vocational flight training', value: 'flightTraining' },
+  { label: 'National test reimbursement', value: 'testReimbursement' },
+  { label: 'Licensing or certification test reimbursement', value: 'licensingReimbursement' },
+  { label: 'Apprentisceship or on-the-job', value: 'apprentisceship' },
```
Comment (akainic) [inline src/js/edu-benefits/utils/options-for-select.js]: Typo: `Apprenticeship`

File: src/js/edu-benefits/utils/options-for-select.js [hunk 3]
```diff
@@ -1595,6 +1595,16 @@ const employmentPeriodTiming = [
...
+  { label: 'Correspondence', value: 'correspondence' },
+  { label: 'Tuition assistance top-off', value: 'tuitionTopUp' }
```
Comment (akainic) [inline src/js/edu-benefits/utils/options-for-select.js]: Is it top-off or top-up?

File: src/js/edu-benefits/utils/validations.js [hunk 4]
```diff
@@ -135,12 +143,13 @@ function isValidDateField(field) {
...
-function dateToMoment(dateField) {
-  return moment({
-    year: dateField.year.value,
-    month: dateField.month.value,
-    day: dateField.day.value
-  });
+function isValidFutureDateField(field) {
+  if (!isBlankDateField(field)) {
+    const momentDate = dateToMoment(field);
+    return momentDate.isValid() && momentDate.year() > 1900;
```
Comment (akainic) [inline src/js/edu-benefits/utils/validations.js]: I'm assuming this isn't just for future dates, but also for recent past dates. Perhaps change the name of the function then?

**Comment by akainic** [comment]
LGTM with a few nits!

File: src/js/edu-benefits/utils/options-for-select.js (hunk 3 above)
Comment (jbalboni) [inline src/js/edu-benefits/utils/options-for-select.js]: Looks like it should be top up, but I'll ask.

**Comment by jbalboni** [comment]
@akainic Should all be fixed now. I can merge it when the CI is done.

**Comment by cweber** [comment]
I am just seeing my name mention above (need to figure out better notifications...).

Yes to the assumption? I was thinking of it as a label to a fieldset. Looks like those are being handled as an H4 in an .input-section on HCA.

But you're right, the framing as a question for a title like that seems odd. Hopefully this can be addressed with better language.

**Comment by jbalboni** [comment]
@cweber Cool. I'll open an issue to change it to an `h4`.

**Comment by cweber** [comment]
More memories! (Hard to recall all of the decisions I made.)

It's already exposed by a conditional field. Even though it is only available to people who select certain fields, it isn't required of those people. ...this gets at another problem Alex and I working on solving which is the "this is important" (put not required) form fields.

**Pull request merged**